### PR TITLE
🐛Garbage collector: possible fix to "spurious" shutdowns

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
One fundamental flaw in the GC was that the order of checking for orphaned services might lead to a wrong closing of these services:
1. Lookup of theoretical services running
2. Lookup of actual services running

That means that if any new project is started between 1 and 2, then the services found in 2 might not be in the previous call to 1. --> the GC then kills these services.
This could be identified by setting the GC Interval to 1 second locally.

This PR inverts this and also refactors slightly this part of the code.
Also the testing was improved hopefully.

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by mguidon — APPROVED** [review]
🤞

**Review by pcrespov — APPROVED** [review]
thx

File: services/web/server/src/simcore_service_webserver/garbage_collector/_core_orphans.py [hunk 1]
```diff
@@ -1,180 +1,124 @@
...
         )
-        for service in running_dynamic_services
-    ]
-    await logged_gather(*tasks, reraise=False)
 
-    _logger.debug("Finished orphaned services removal")
+        # compute the difference to find the orphaned services
+        orphaned_running_service_ids = (
+            set(running_services_by_id) - potentially_running_service_ids_set
+        )
```
Comment (bisgaard-itis) [inline services/web/server/src/simcore_service_webserver/garbage_collector/_core_orphans.py:114]: What happens if a service is running but then closed before `potentially_running_service_ids_set` is computed? I.e. a service is marked as orphaned but is actually closed. Will the GC complain when trying to close a service which is already closed?

**Review by bisgaard-itis — APPROVED** [review]

**Review by bisgaard-itis — APPROVED** [review]
This is definitely a nice catch regardless of whether or not it will resolve all the spurious shutdowns.

**Review by odeimaiz — COMMENTED** [review]
Merci!

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks!

File: services/web/server/src/simcore_service_webserver/garbage_collector/_core_orphans.py (hunk 1 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/garbage_collector/_core_orphans.py:114]: I think not. but I will check. good point

**Pull request merged**